go-controller/pkg/util/ovs.go

```diff
@@ -90,17 +91,25 @@ var MetricOvnCliLatency *prometheus.HistogramVec
 var maxArgs int
 
 func init() {
-	out, err := exec.Command("getconf", "ARG_MAX").Output()
-	if err != nil {
-		v, err := strconv.Atoi(strings.TrimSuffix(string(out), "\n"))
-		if err == nil {
-			maxArgs = v
-		}
+	maxArgs = findMaxArgsUsable(defaultOSMaxArgs)
+	klog.Infof("Maximum command line arguments set to: %d", maxArgs)
+}
+
+// findMaxArgsUsable finds the maximum amount of usable args on the system, which may be
+// different than what the kernel returns for ARG_MAX
+func findMaxArgsUsable(estimatedMax int) int {
+	backoff := .9
+	args := make([]string, estimatedMax)
+	for i := range args {
+		args[i] = "a"
+	}
+	if estimatedMax <= minOSArgs {
+		return minOSArgs
 	}
-	if maxArgs == 0 {
-		klog.Warningf("Unable to detect OS MAX_ARGS, defaulting to: %d", defaultOSMaxArgs)
-		maxArgs = defaultOSMaxArgs
+	if _, err := exec.Command("/bin/true", args...).Output(); err != nil {
+		return findMaxArgsUsable(int(float64(estimatedMax) * backoff))
```

Comment:
worst case stack depth with current parameters would be 52.
in each you are allocating a slice of size `estimatedMax` bytes so memory consumption would stack up.

why not just do a while (for) loop ?